run-shell/runner.go

```diff
@@ -0,0 +1,155 @@
+package runner
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/google/shlex"
+	"github.com/mitchellh/mapstructure"
+	"github.com/stoic-cli/stoic-cli-core"
+	"github.com/stoic-cli/stoic-cli-core/tool"
+)
+
+func NewRunner(s stoic.Stoic, tool stoic.Tool) (tool.Runner, error) {
+	var options RunnerOptions
+	err := mapstructure.Decode(tool.Config().Runner.Options, &options)
+	if err != nil {
+		return nil, err
+	}
+
+	if options.SetupEnvironment == nil {
+		options.SetupEnvironment = map[string]string{}
+	}
+	if options.SetupParameters == nil {
+		options.SetupParameters = map[string]interface{}{}
+	}
+	if options.Environment == nil {
+		options.Environment = map[string]string{}
+	}
+	if options.Parameters == nil {
+		options.Parameters = map[string]interface{}{}
+	}
+
+	return Runner{s, options}, nil
+}
+
+type RunnerOptions struct {
```

Comment:
type name will be used as runner.RunnerOptions by other packages, and that stutters; consider calling this Options